ui/lib/sync/addon/components/secrets/page/destinations/destination/sync.hbs

```diff
@@ -17,7 +17,7 @@
 <form {{on "submit" (perform this.setAssociation)}}>
   <MessageError @errorMessage={{this.error}} />
 
-  {{#if this.syncedSecret}}
+  {{#if (and this.syncedSecret this.mountPath)}}
```

Comment:
Curious why `mountPath` was added. Wouldn't the banner show after a secret is synced, be hidden if the mount path was removed and then shown again if a new mount path is selected but before a second secret was synced with this addition?

Reply:
Good catch! It's because if a user clears the mountPath then the secret link to in the success banner errors. But I think a better way to handle it is to just clear `syncedSecret` when mountPath is cleared so that essentially the form resets when the search select `X` is clicked
<img width="1064" alt="Screenshot 2024-01-09 at 4 43 23 PM" src="https://github.com/hashicorp/vault/assets/68122737/159b41e6-c67e-4904-be63-3b315fea27e1">

Reply:
Another option is to store the `syncedSecret` info in an object with the secretPath and mountPath if you think the banner remaining is preferred

Reply:
Ahhh gotcha. I think clearing `syncedSecret` on mount path change should be fine. At that point the user is aware that they can sync another secret if they are making changes.

Reply:
Okay cool! Thanks for the feedback.